sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java

```diff
@@ -757,16 +761,16 @@ public void testSelectStarFromSelectSingleColumnWithLimitDescending() throws Exc
             newScanQueryBuilder()
                 .dataSource(CalciteTests.DATASOURCE1)
                 .intervals(querySegmentSpec(Filtration.eternity()))
-                .columns(ImmutableList.of("__time", "dim1"))
+                .columns(ImmutableList.of("dim1"))
                 .limit(2)
-                .order(ScanQuery.Order.DESCENDING)
+                .order(ScanQuery.Order.NONE)
```

Comment:
Added a comment, and an additional query within this test that uses the outer context limit instead. One difference there is that the new query includes __time within the results (otherwise the scan query rejects the query, since it is ordering on __time)